lib/inky.rb

```diff
@@ -2,46 +2,58 @@
 require_relative 'inky/configuration'
 require_relative 'inky/component_factory'
 
+Dir.glob(File.expand_path("../inky/components/*", __FILE__)).each do |file|
+  require_relative file
+end
+
 module Inky
   class Core
-    attr_accessor :components, :column_count, :component_lookup, :component_tags
+    attr_accessor :components, :column_count, :component_tags
+    
+    # These constants are used to circumvent an issue with JRuby Nokogiri.
+    # For more details see https://github.com/zurb/inky-rb/pull/94
+    INTERIM_TH_TAG = 'inky-interim-th'.freeze
+    INTERIM_TH_TAG_REGEX = %r{(?<=\<|\<\/)#{Regexp.escape(INTERIM_TH_TAG)}}
+
+    DEFAULT_COMPONENTS = {
+      "button" => Inky::Components::Button,
+      "row" => Inky::Components::Row,
+      "inky" => Inky::Components::Inky,
+      "columns" => Inky::Components::Columns,
+      "container" => Inky::Components::Container,
+      "block-grid" => Inky::Components::BlockGrid,
+      "menu" => Inky::Components::Menu,
+      "center" => Inky::Components::Center,
+      "callout" => Inky::Components::Callout,
+      "spacer" => Inky::Components::Spacer,
+      "wrapper" => Inky::Components::Wrapper,
+      "item" => Inky::Components::MenuItem
+    }
 
     include ComponentFactory
-    def initialize(options = {})
-      self.components = {
-        button: 'button',
-        row: 'row',
-        columns: 'columns',
-        container: 'container',
-        inky: 'inky',
-        block_grid: 'block-grid',
-        menu: 'menu',
-        center: 'center',
-        callout: 'callout',
-        spacer: 'spacer',
-        wrapper: 'wrapper',
-        menu_item: 'item'
-      }.merge(options[:components] || {})
 
-      self.component_lookup = components.invert
+    def initialize(options = {})
+      self.components = DEFAULT_COMPONENTS
+        .merge(options[:components] || ::Inky.configuration.components)
+        .transform_values { |component_class| component_class.new(self) }
```

Comment:
`transform_values` is Ruby 2.4+ only; we support older rubies (I think). Would be easy to `require 'backports/2.4.0/hash/transform_values'` or change the code.